### ISSUE TITLE
sphinx references

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/picongpu#3054]
Sphinx supports a single definition of a reference and citing it in any `*.rst` file.
This allows citing e.g. [**FLYCHK**](https://picongpu.readthedocs.io/en/latest/models/collisional_ionization.html#flychk) from within [the warm copper](https://picongpu.readthedocs.io/en/latest/examples/WarmCopper/README.html) without having to redefine. 

**Pro:**
 - only one consistent definition of a reference
 - avoids various warnings from sphinx (see #2564)

**Con:**
 - definition will not necessarily be found at the bottom of a page 

@ComputationalRadiationPhysics/picongpu-developers What do you think of this?

**Comment by ax3l** [comment]
Yes, I am aware of that but preferred double-definitions since this keeps in HTML all references local instead of a gigantic one somewhere at the end of the HTML index.

**Comment by PrometheusPi** [comment]
It would be nice to have a list of references (which is not compiled into a gigantic html page in the end) and add each reference from that list for each `*.rst` page similar to what LaTeX does with `\cite{}` , `\bibliography{}` and `ref.bib`-files. I will have look into it, whether this is possible with sphinx. 

But in general I agree: I also prefer a possible redefinition on various pages instead of huge bibliography in the end, where context is mixed.

**Comment by PrometheusPi** [comment]
This project might solve this issue: https://github.com/mcmtroffaes/sphinxcontrib-bibtex
(But so far I did not find a good multi-page, multi-citation example to check whether bibliographies contain only locally cited entries or test it myself.)

**Comment by ax3l** [comment]
Makes sense as long as we can include those references with each page.

**Comment by PrometheusPi** [comment]
I will test that.

**Comment by sbastrakov** [comment]
@PrometheusPi is this solved already? Sorry, I got confused with all the references stuff, just remember there were some PRs on this.

**Comment by PrometheusPi** [comment]
@sbastrakov Sorry - had no time to test this so far.

**Comment by sbastrakov** [comment]
@PrometheusPi ah, sorry, I thought maybe it's already implemented and the issue just left open.

**Comment by ax3l** [comment]
>  You can probably just give the references unique names to get rid of the warnings, e.g. by pre-fixing their aliases with the current section.

Can be fixed with double `_` in `.rst` references.
